servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/AbstractTimeoutHttpFilter.java

```diff
@@ -151,6 +158,64 @@ private Executor contextExecutor(final HttpRequestMetaData requestMetaData,
                 context.executor() : context.ioExecutor();
     }
 
+    // This is responsible for ensuring that we don't abandon the response resources due to the use of Single.timeout.
+    // It does so by retaining a reference to the StreamingHttpResponse in case we receive a cancellation. It will
+    // then attempt to drain the resource. We're protected from multiple subscribes because the TimeoutSingle will
+    // short circuit also on upstream cancellation, although that is not obviously correct behavior.
+    private static final class CleanupSubscriber implements SingleSource.Subscriber<StreamingHttpResponse> {
+
+        private static final AtomicReferenceFieldUpdater<CleanupSubscriber, Object> stateUpdater =
+                AtomicReferenceFieldUpdater.newUpdater(CleanupSubscriber.class, Object.class, "state");
+
+        private static final Object COMPLETE = "cancelled";
+
+        private final SingleSource.Subscriber<? super StreamingHttpResponse> delegate;
+        @SuppressWarnings("unused")
+        @Nullable
+        private volatile Object state;
+
+        CleanupSubscriber(final SingleSource.Subscriber<? super StreamingHttpResponse> delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public void onSubscribe(Cancellable cancellable) {
+            delegate.onSubscribe(() -> {
+                onCancel();
+                cancellable.cancel();
```

Comment:
To be safe, consider using `try-finally` block as we do in other places, like `BeforeFinallyHttpOperator`